src/beamlime/services/dashboard_plotly.py

```diff
@@ -0,0 +1,176 @@
+    AdaptingMessageSource,
+    ChainedAdapter,
+    Da00ToScippAdapter,
+    KafkaToDa00Adapter,
+)
+from beamlime.kafka.source import KafkaMessageSource
+
+control_config = load_config(namespace='monitor_data')['control']
+config_service = ConfigService(kafka_config=control_config)
+config_service_thread = threading.Thread(target=config_service.start)
+config_service_thread.start()
+
+app = Dash(__name__)
+
+# Initialize empty plot storage
+monitor_plots: dict[str, go.Figure] = {}
+detector_plots: dict[str, go.Figure] = {}
+
+app.layout = html.Div(
+    [
+        html.Div(
+            [
+                html.Label('Update Speed (ms)'),
+                dcc.Slider(
+                    id='update-speed',
+                    min=8,
+                    max=13,
+                    step=0.5,
+                    value=10,
+                    marks={i: {'label': f'{2**i}'} for i in range(8, 14)},
+                ),
+                html.Label('Time-of-arrival bins'),
+                dcc.Slider(
+                    id='num-points',
+                    min=10,
+                    max=500,
+                    step=10,
+                    value=100,
+                    marks={i: str(i) for i in range(0, 501, 100)},
+                ),
+                html.Button('Clear', id='clear-button', n_clicks=0),
+            ],
+            style={'width': '300px', 'float': 'left', 'padding': '10px'},
+        ),
+        html.Div(id='plots-container', style={'margin-left': '320px'}),
+        dcc.Interval(
+            id='interval-component',
+            interval=200,  # in milliseconds
+            n_intervals=0,
+        ),
+    ]
+)
+
+
+def create_detector_plot(key: str) -> go.Figure:
+    """Create a new Plotly figure for a detector."""
+    fig = go.Figure()
+    fig.add_heatmap(z=[[1, 2], [3, 4]], colorscale='Viridis')
+    # Setting uirevision ensures that the figure is not redrawn on every update, keeping
+    # the zoom level and position.
+    fig.update_layout(title=key, width=500, height=400, uirevision=key)
+    return fig
+
+
+def create_monitor_plot(key: str, data: sc.DataArray) -> go.Figure:
+    """Create a new Plotly figure for a monitor."""
+    fig = go.Figure()
+    fig.add_scatter(x=[], y=[], mode='lines', line_width=2)
+    # Setting uirevision ensures that the figure is not redrawn on every update, keeping
+    # the zoom level and position.
+    dim = data.dim
+    fig.update_layout(
+        title=key,
+        width=500,
+        height=400,
+        xaxis_title=f'{dim} [{data.coords[dim].unit}]',
+        yaxis_title=f'[{data.unit}]',
+        uirevision=key,
+    )
+    return fig
+
+
+consumer_config = load_config(namespace='visualization')['consumer']
+consumer_kafka_config = consumer_config['kafka']
+consumer_kafka_config['group.id'] = 'monitor_data_dashboard'
+
+consumer = kafka_consumer.make_bare_consumer(
+    topics=topic_for_instrument(topic=consumer_config['topics'], instrument='dummy'),
```

Comment:
We already talked about this but leaving a memo here.
This should also be configurable.

Reply:
Fixed.